profiling/src/lib.rs

```diff
@@ -475,13 +479,19 @@ extern "C" fn rinit(_type: c_int, _module_number: c_int) -> ZendResult {
         output_pprof,
     ) = unsafe {
         let profiling_enabled = config::profiling_enabled();
+        let profiling_experimental_features_enabled =
+            config::profiling_experimental_features_enabled();
         (
             profiling_enabled,
+            profiling_enabled && profiling_experimental_features_enabled,
             profiling_enabled && config::profiling_endpoint_collection_enabled(),
-            profiling_enabled && config::profiling_experimental_cpu_time_enabled(),
+            profiling_experimental_features_enabled
+                || profiling_enabled && config::profiling_experimental_cpu_time_enabled(),
             profiling_enabled && config::profiling_allocation_enabled(),
-            profiling_enabled && config::profiling_experimental_timeline_enabled(),
-            profiling_enabled && config::profiling_experimental_exception_enabled(),
+            profiling_experimental_features_enabled
+                || profiling_enabled && config::profiling_experimental_timeline_enabled(),
+            profiling_experimental_features_enabled
+                || profiling_enabled && config::profiling_experimental_exception_enabled(),
```

Comment:
the checks should either read `profiling_enabled && (profiling_experimental_features_enabled || config::profiling_experimental_..._enabled())`, or, I'd prefer: ` let profiling_experimental_features_enabled =
            profiling_enabled && config::profiling_experimental_features_enabled();`.

Reply:
Yes, you are right, I messed the logic up.